### PR TITLE
Converted HC to make sure that all the GO xrefs have an evidence

### PR DESCRIPTION
**Pull request by thomasmaurel**

### CODE REVIEW

File: lib/Bio/EnsEMBL/DataCheck/Checks/EvidenceFreeGO.pm [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME        => 'EvidenceFreeGO',
```
Comment (james-monkeyshines) [inline lib/Bio/EnsEMBL/DataCheck/Checks/EvidenceFreeGO.pm]: This has always struck me as an odd name for the HC...
```suggestion
  NAME        => 'GOXrefEvidence',
```

File: lib/Bio/EnsEMBL/DataCheck/Checks/EvidenceFreeGO.pm [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  DESCRIPTION => 'All GO xrefs have an evidence',
+  GROUPS         => ['compare_core', 'xref'],
```
Comment (james-monkeyshines) [inline lib/Bio/EnsEMBL/DataCheck/Checks/EvidenceFreeGO.pm]:
```suggestion
  GROUPS         => ['xref'],
```

File: lib/Bio/EnsEMBL/DataCheck/index.json [hunk 3]
```diff
@@ -538,6 +538,16 @@
       "name" : "EpigenomeHasSegmentationFile",
       "package_name" : "Bio::EnsEMBL::DataCheck::Checks::EpigenomeHasSegmentationFile"
    },
+   "EvidenceFreeGO" : {
+      "datacheck_type" : "critical",
+      "description" : "All GO xrefs have an evidence",
+      "groups" : [
+         "compare_core",
```
Comment (james-monkeyshines) [inline lib/Bio/EnsEMBL/DataCheck/index.json]:
```suggestion
```

File: lib/Bio/EnsEMBL/DataCheck/Checks/EvidenceFreeGO.pm (hunk 1 above)
Comment (thomasmaurel) [reply lib/Bio/EnsEMBL/DataCheck/Checks/EvidenceFreeGO.pm]: Sure, make sense. Done in commit

File: lib/Bio/EnsEMBL/DataCheck/index.json (hunk 3 above)
Comment (thomasmaurel) [reply lib/Bio/EnsEMBL/DataCheck/index.json]: Thanks, that was a copy paste, erm... Fixed in commit.

**Pull request merged**